pkg/hypershift/hypershift.go

```diff
@@ -141,6 +142,11 @@ func ParseHostedControlPlane(hcp *unstructured.Unstructured) (*HostedControlPlan
 		return nil, fmt.Errorf("failed to extract controllerAvailabilityPolicy: %v", err)
 	}
 
+	controlPlanePriorityClassAnnotation, _, err := unstructured.NestedString(hcp.UnstructuredContent(), "metadata", "annotations", "hypershift.openshift.io/control-plane-priority-class")
```

Comment:
nit: it'd be nice to have a comment linking the counterpart source code. So when someone with less context reads the context can understand where this is coming from.